src/client/Client.cc

```diff
@@ -11742,6 +11743,14 @@ size_t Client::_vxattrcb_dir_rctime(Inode *in, char *val, size_t size)
   return snprintf(val, size, "%ld.09%ld", (long)in->rstat.rctime.sec(),
       (long)in->rstat.rctime.nsec());
 }
+bool Client::_vxattrcb_dir_pin_exists(Inode *in)
+{
+  return in->dir_pin != -ENODATA;
```

Comment:
RANK_NONE

Reply:
not found RANK_NONE? you mean MDS_RANK_NONE?

Reply:
If we use MDS_RANK_NONE at both places, and even if ceph-fuse and mds are all new version, but directory is not set any pin value,  getfattr ceph.dir.pin will always return "No such attribute". 

If we use ENODATA at both places,  new version of ceph-fuse and mds will return -1, but new version of ceph-fuse and old version of mds will still return "No such attribute".

Reply:
@ukernel  Pls check my comments again, thanks.